File: test/vuln.php
<?php

$user_data = $_POST['data'];

// Safe
$json = json_decode($user_data);

// Vulnerable (High Confidence)
$obj = unserialize($user_data);

// Hardcoded / Safe (Low Confidence - Ignored/Blue)
$test = unserialize('s:4:"test";');
$test2 = unserialize("a:1:{s:3:'foo';s:3:'bar';}");

File: test/vuln.py
import pickle
import yaml
import json

def load_data(data):
    # Safe
    print(json.loads(data))
    
    # Vulnerable: Pickle
    # POTENTIAL RCE
    obj = pickle.load(data)
    
    # Vulnerable: PyYAML
    config = yaml.load(data)
    
    # Safe YAML
    safe_config = yaml.safe_load(data)
